### PR TITLE
Fix #1956 bad type usage on 32-bit

### LINKED ISSUE #1956: Memory error on 32-bit build (zstd)
Runnin test suite on 32-bit (i686, armv7hl)
https://koji.fedoraproject.org/koji/taskinfo?taskID=64552748

```
Note: these tests might take up to a minute. Don't worry :-)
Using PHP version 8.0.4RC1 (32 bits)
Testing class Redis
testMinimumVersion                                             [PASSED]
...
testCompressionLZF                                             [PASSED]
testCompressionZSTD                                            
Fatal error: Possible integer overflow in memory allocation (4294967294 + 4096) in /builddir/build/BUILD/php-pecl-redis5-5.3.4/NTS/tests/RedisTest.php on line 4581
```

5.3.3 was OK

### PR DESCRIPTION
**Pull request by remicollet**
See #1956

### CODE REVIEW

File: redis_array.c [hunk 1]
```diff
@@ -224,7 +224,7 @@ PHP_METHOD(RedisArray, __construct)
     RedisArray *ra = NULL;
     zend_bool b_index = 0, b_autorehash = 0, b_pconnect = 0, consistent = 0;
     HashTable *hPrev = NULL, *hOpts = NULL;
-    long l_retry_interval = 0;
+    zend_long l_retry_interval = 0;
```
Comment (remicollet) [inline redis_array.c:227]: else bad pointer conversion in `redis_conf_long` call

File: redis_array.c [hunk 2]
```diff
@@ -276,7 +276,7 @@ PHP_METHOD(RedisArray, __construct)
     }
 
     ra = ra_make_array(Z_ARRVAL_P(z0), &z_fun, &z_dist, hPrev, b_index,
-                       b_pconnect, l_retry_interval, b_lazy_connect,
+                       b_pconnect, (long)l_retry_interval, b_lazy_connect,
```
Comment (remicollet) [inline redis_array.c:279]: Hack... will be better (but a much biger change) to use zend_long everywhere.

So used to keep this PR minimal

File: library.c [hunk 3]
```diff
@@ -2968,7 +2968,7 @@ redis_unpack(RedisSock *redis_sock, const char *val, int val_len, zval *z_ret)
 #ifdef HAVE_REDIS_ZSTD
             {
                 char *data;
-                size_t len;
+                unsigned long long len;
```
Comment (remicollet) [inline library.c:2971]: From zstd.h

```
#define ZSTD_CONTENTSIZE_UNKNOWN (0ULL - 1)
#define ZSTD_CONTENTSIZE_ERROR   (0ULL - 2)
ZSTDLIB_API unsigned long long ZSTD_getFrameContentSize(const void *src, size_t srcSize);

```

So we have to use an `unsigned long long` (perhaps a cast `(size_t)ZSTD_CONTENTSIZE_UNKNOWN` may also work)

**Comment by remicollet** [comment]
Fedora build and test suite OK on all arches (using this patch)
https://koji.fedoraproject.org/koji/taskinfo?taskID=64555141

**Comment by remicollet** [comment]
@michael-grunder please review as related to c48b3a2ba3d677efceb9ac1958246631f6bd8037

**Comment by michael-grunder** [comment]
Thank you @remicollet!

If you're OK with 6da1758 I'll get this merged.

Edit:  We should also do 32-bit tests.  I'll make sure to add them when moving to GitHub Actions

**Comment by remicollet** [comment]
> If you're OK with [6da1758](https://github.com/phpredis/phpredis/commit/6da1758af83380990b4afc684652febfc873904e) I'll get this merged.

LGTM

**Pull request merged**